### PR TITLE
refactor: set openapi server url as close to startup as possible

### LINKED ISSUE #248: refactor setting of - Server.OpenApiSpec.Servers
Right now we do it in the constructor of [Server function](https://github.com/go-fuego/fuego/blob/main/server.go#L14). We should push this closer to startup of the server as we can interpret if the user is running an http or https server as well as get the proper address since #245 we do not know the final address we use until `s.setup()` is called nor if the server is https or http. 

In regards to just printing. We can just assume `http` for now. If we find we have a reason to allow the user to ensure the server setup is https we can update the `s.OutputOpenAPISpec()` to be `s.OutputOpenAPISpec(isTLS bool)`. We really don't have another way of doing this since we truly don't know if the server is an http or https webserver until startup.

### PR DESCRIPTION
**Pull request by dylanhitt**
Closes #248 

Will conflict with https://github.com/go-fuego/fuego/pull/245 but it does solve the issue of creating the server too early so that code could be simplified now I believe. 

Thanks

### CODE REVIEW

File: openapi.go [hunk 1]
```diff
@@ -95,6 +95,11 @@ func declareAllTagsFromOperations(s *Server) {
 // Also serves a Swagger UI.
 // To modify its behavior, use the [WithOpenAPIConfig] option.
 func (s *Server) OutputOpenAPISpec() openapi3.T {
+	s.OpenAPI.Description().Servers = append(s.OpenAPI.Description().Servers, &openapi3.Server{
+		URL:         s.proto() + "://" + s.Addr,
```
Comment (ccoVeille) [inline openapi.go:99]: there should be a method on `Server` for computing `s.proto() + "://" + s.Addr`

**Review by ccoVeille — APPROVED** [review]

**Review by EwenQuim — APPROVED** [review]
Nice, thanks!

File: openapi.go (hunk 1 above)
Comment (ccoVeille) [reply openapi.go:99]: It's also used in

https://github.com/go-fuego/fuego/blob/main/serve.go

Comment (dylanhitt) [reply openapi.go:99]: OIC in the startup messages. Will update in a second

**Comment by dylanhitt** [comment]
Updated with `s.url()` method.

File: openapi.go (hunk 1 above)
Comment (ccoVeille) [reply openapi.go]: Adding this would have been appreciated
```
    Co-authored-by: ccoVeille <3875889+ccoVeille@users.noreply.github.com>
```

Comment (dylanhitt) [reply openapi.go]: I don't have push rights to main. If @EwenQuim wouldn't mind doing it that would be great.

**Pull request merged**